csharp/extractor/Semmle.Extraction.CSharp/Entities/Expressions/ImplicitToString.cs

```diff
@@ -14,14 +14,21 @@ internal sealed class ImplicitToString : Expression
         /// </summary>
         private static IMethodSymbol? GetToStringMethod(ITypeSymbol? type)
         {
-            return type?
+            if (type is null)
+            {
+                return null;
+            }
+
+            var toString = type
                 .GetMembers()
                 .OfType<IMethodSymbol>()
                 .Where(method =>
                     method.GetName() == "ToString" &&
                     method.Parameters.Length == 0
                 )
                 .FirstOrDefault();
+
+            return toString ?? GetToStringMethod(type.BaseType);
```

Comment:
It's odd that we have no helper for getting the base types already. Maybe adding this to `SymbolExtensions` would make sense.

Reply:
It looks like the base types are used in all sorts of different ways and I didn't find any overlapping use cases for recursively going through the type hierarchy to find a method.